### PR TITLE
docker in docker isolation

### PR DESCRIPTION
**Pull request by mhshami01**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -55,4 +46,8 @@ RUN \
     echo "\n# add local python programs to PATH\nexport PATH=$PATH:$HOME/.local/bin\n" >> $HOME/.bashrc && \
     # enable some useful aliases
     sed -i -e "s/#alias/alias/" $HOME/.bashrc && \
-    echo "\n# skip sudo for docker commands\nalias docker='sudo docker'" >> $HOME/.bashrc
+    pip install cookiecutter
```
Comment (marianan) [inline .devcontainer/Dockerfile:49]: >cookiecutter [](http://example.com/codeflow?start=16&length=12)

why is cookiecutter needed here?

File: .devcontainer/library-scripts/docker-in-docker-debian.sh [hunk 2]
```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+#
+# Docs: https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker-in-docker.md
+# Maintainer: The VS Code and Codespaces Teams
+#
+# Syntax: ./docker-in-docker-debian.sh [enable non-root docker access flag] [non-root user] [use moby]
```
Comment (marianan) [inline .devcontainer/library-scripts/docker-in-docker-debian.sh:10]: >docker-in-docker-debian [](http://example.com/codeflow?start=12&length=23)

is this file any different from the ones we have int he other tools devcontainers?

**Review by marianan — CHANGES_REQUESTED** [review]
:clock1:

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,38 +1,33 @@
...
+ARG PYTHON_VERSION="3.9"
 
-# see https://hub.docker.com/_/python/?tab=tag
-FROM python:${PYTHON_VERSION}-buster
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${PYTHON_VERSION}
 
-# Dependencies versions
-ARG AZURE_CLI_VERSION="2.34.1-1~buster"
+# Other Dependencies versions
+ARG AZURE_CLI_VERSION="2.34.1-1~bullseye"
```
[the remark this answers is not in the document]
Comment (mhshami01) [reply .devcontainer/Dockerfile:7]: because we are using a different image - from a Microsoft ACR

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (mhshami01) [reply .devcontainer/Dockerfile:49]: When running some manual tests, I used iotedgedev to setup a solution.

File: .devcontainer/library-scripts/docker-in-docker-debian.sh (hunk 2 above)
Comment (mhshami01) [reply .devcontainer/library-scripts/docker-in-docker-debian.sh:10]: it is identical

**Review by marianan — APPROVED** [review]
:shipit:

**Pull request merged**